projects/hslayers/src/components/add-data/url/arcgis/arcgis.service.ts

```diff
@@ -258,20 +258,27 @@ export class HsUrlArcGisService implements HsUrlTypeServiceModel {
    */
   async calcAllLayersExtent(
     layers: ArcGISRestResponseLayer[],
-    options: addLayerOptions
+    options: addLayerOptions,
   ) {
     try {
       const layersCaps = await Promise.all(
         layers.map(async (l) => {
           return await this.hsArcgisGetCapabilitiesService.request(
-            `${this.data.get_map_url}/${l.id}`
+            `${this.data.get_map_url}/${l.id}`,
           );
-        })
+        }),
       );
-      const layersExtents = layersCaps.map((l) => {
-        const extent = l.response.extent;
-        return this.transformLayerExtent(extent, this.data);
-      });
+      const layersExtents = layersCaps
+        .map((l) => {
+          if (
+            Object.values(l.response.extent)
+              .filter((v) => !(v instanceof Object))
```

Comment:
```suggestion
              .filter((v) => !isNaN(v)
```

In a rare scenario, a number can be an instance of Object (Number initialized via the new Number())! It probably cannot happen in this case, but I feel using this comparison is generally unsafe. Additionally, it will interpret the `undefined` and `null` values also as "not instanceof Object" which is undesired. I would prefer isNaN comparison for these reasons. It will still fail for `Date` and `null`, but will solve the `undefined`. See this fiddle: https://jsfiddle.net/42ecxmkb/

Reply:
Thanks for pointing it out. It really was not a good  https://github.com/hslayers/hslayers-ng/pull/4182/commits/2731938e8f48c4c8aeb46957440b34e45b9a79c4